tests/contrib/minikube/test_kubernetes_executor.py

```diff
@@ -33,64 +34,164 @@
+                                  timeout):
+        tries = 0
+        state = ''
+        max_tries = max(int(timeout / 5), 1)
+        # Wait 100 seconds for the operator to complete
+        while tries < max_tries:
+            time.sleep(5)
+
+            # Trigger a new dagrun
+            result = requests.get(
+                'http://{host}/api/experimental/dags/{dag_id}/'
+                'dag_runs/{execution_date}/status'
+                .format(host=host,
+                        dag_id=dag_id,
+                        execution_date=execution_date)
+            )
+            print(result)
+            self.assertEqual(result.status_code, 200, "Could not get the status")
+            result_json = result.json()
+            print(result_json)
+            state = result_json['state']
+            check_call(
+                ["echo", "Attempt {}: Current state of dag is {}".format(tries, state)])
+            print("Attempt {}: Current state of dag is {}".format(tries, state))
+
+            if state == expected_final_state:
+                break
+            tries += 1
+
+        self.assertEqual(state, expected_final_state)
+
+        # Maybe check if we can retrieve the logs, but then we need to extend the API
 
-        #  Enable the dag
+    def start_dag(self, dag_id, host):
         result = requests.get(
-            'http://{}/api/experimental/'
-            'dags/example_python_operator/paused/false'.format(host)
+            'http://{host}/api/experimental/'
+            'dags/{dag_id}/paused/false'.format(host=host, dag_id=dag_id)
         )
-        self.assertEqual(result.status_code, 200, "Could not enable DAG")
+        self.assertEqual(result.status_code, 200, "Could not enable DAG: {result}"
+                         .format(result=result.reason))
 
         # Trigger a new dagrun
         result = requests.post(
-            'http://{}/api/experimental/'
-            'dags/example_python_operator/dag_runs'.format(host),
+            'http://{host}/api/experimental/'
+            'dags/{dag_id}/dag_runs'.format(host=host, dag_id=dag_id),
             json={}
         )
-        self.assertEqual(result.status_code, 200, "Could not trigger a DAG-run")
+        self.assertEqual(result.status_code, 200, "Could not trigger a DAG-run: {result}"
+                         .format(result=result.reason))
 
         time.sleep(1)
 
         result = requests.get(
             'http://{}/api/experimental/latest_runs'.format(host)
         )
-        self.assertEqual(result.status_code, 200, "Could not get the latest DAG-run")
+        self.assertEqual(result.status_code, 200, "Could not get the latest DAG-run:"
+                                                  " {result}"
+                         .format(result=result.reason))
         result_json = result.json()
+        return result_json
+
+    def test_integration_run_dag(self):
+        host_ip = check_output(['minikube', 'ip'])
```

Comment:
This part of getting the host is now defined twice, maybe create a separatew function for this?

Reply:
Sure, I'll do that now